tests/repo_finder/test_commit_finder.py

```diff
@@ -0,0 +1,112 @@
+    assert matched_tags
+    assert matched_tags[0] == target_tag
+
+
+input_pattern = re.compile(r"[0-9]{1,3}(\.[0-9a-z]{1,3}){,5}([-+#][a-z0-9].+)?", flags=re.IGNORECASE)
+
+
+@given(text())
+@settings(max_examples=1000)
+def test_pattern_generation(version: str) -> None:
+    """Test stability of pattern creation from user input."""
+    # pylint: disable=protected-access
+    # Try creating a PURL from the version, if successful pass the purl.version to commit finder's pattern builder.
+    if not version:
+        return
+    try:
+        purl = PackageURL(name="test", version=version, type="maven")
+        if not purl.version:
+            return
+    except ValueError as error:
+        logger.debug(error)
+        return
+
+    commit_finder._build_version_pattern(purl.name, purl.version)
+    assert True
+
+
+# These numbers should be kept low as the complex regex makes generation slow.
+VERSION_ITERATIONS = 50  # The number of times to iterate the test_version_to_tag_matching test.
+TAG_ITERATIONS = 10  # The number of tags to generate per version iteration.
+
+
+@given(data())
+@settings(max_examples=VERSION_ITERATIONS, deadline=None)
+def test_version_to_tag_matching(_data: DataObject) -> None:  # noqa: PT019
+    """Test matching generated versions to generated tags.
+
+    This test verifies that a similar version and tag can be matched by the commit finder.
+    """
+    # pylint: disable=protected-access
+    # Generate the version
+    version = _data.draw(hypothesis.strategies.from_regex(input_pattern, fullmatch=True))
+    if not version:
+        return
+    try:
+        purl = PackageURL(name="test", version=version, type="maven")
+        if not purl.version:
+            return
+        # Build the pattern from the version.
+        pattern, parts, _ = commit_finder._build_version_pattern(purl.name, purl.version)
+        if not pattern:
+            return
+        # Generate the tag from a pattern that is very similar to how version patterns are made.
+        sep = "[^a-z0-9]"
+        tag_pattern = (
+            "(?P<prefix_0>(?:[a-z].*(?:[a-z0-9][a-z][0-9]+|[0-9][a-z]|[a-z]{2}))|[a-z]{2})?("
+            "?P<prefix_sep_0>(?:(?:(?<![0-9a-z])[vrc])|(?:[^0-9a-z][vrc])|[^0-9a-z])(?:[^0-9a-z])?)?"
+        )
+        for count, part in enumerate(parts):
+            if count > 0:
+                tag_pattern = tag_pattern + f"{sep}"
+            tag_pattern = tag_pattern + part
+        tag_pattern = tag_pattern + f"({sep}[a-z].*)?"
+        compiled_pattern = re.compile(tag_pattern, flags=re.IGNORECASE)
+        # Generate tags to match the generated version.
+        for _ in range(TAG_ITERATIONS):
+            tag = _data.draw(hypothesis.strategies.from_regex(compiled_pattern, fullmatch=True))
+            # Perform the match.
+            match = pattern.match(tag)
+            assert match
+    except ValueError as error:
+        logger.debug(error)
```

Comment:
This try-except block seems too broad and I'm not sure I understand it. Which statements did you expect to throw a `ValueError`?